pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -429,22 +430,23 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication, VaultGuard {
         _takeDebt(wrappedToken, amountWrappedRaw);
 
         // Update buffer balances.
-        bytes32 bufferBalances = _bufferTokenBalances[wrappedToken];
-        bufferBalances = PackedTokenBalance.toPackedBalance(
-            bufferBalances.getBalanceRaw() + amountUnderlyingRaw,
-            bufferBalances.getBalanceDerived() + amountWrappedRaw
-        );
-
-        _bufferTokenBalances[wrappedToken] = bufferBalances;
+        _bufferTokenBalances[wrappedToken] = PackedTokenBalance.toPackedBalance(amountUnderlyingRaw, amountWrappedRaw);
 
-        // Amount of shares to issue is the total underlying token that the user is depositing.
+        // At initialization, the initial "BPT rate" is 1, so the `issuedShares` is simply the sum of the initial
+        // buffer token balances, converted to underlying.
         issuedShares = wrappedToken.convertToAssets(amountWrappedRaw) + amountUnderlyingRaw;
-
         _ensureMinimumTotalSupply(issuedShares);
 
+        // Divide `issuedShares` between the zero address, which receives the minimum supply, and the account
+        // depositing the tokens to initialize the buffer, which receives the balance.
+        _bufferTotalShares[wrappedToken] = _MINIMUM_TOTAL_SUPPLY;
+        _bufferLpShares[wrappedToken][address(0)] = _MINIMUM_TOTAL_SUPPLY;
+
+        emit BufferSharesMinted(wrappedToken, address(0), _MINIMUM_TOTAL_SUPPLY);
+
         issuedShares -= _MINIMUM_TOTAL_SUPPLY;
+
         _mintBufferShares(wrappedToken, sharesOwner, issuedShares);
-        _mintMinimumBufferSupplyReserve(wrappedToken);
```

Comment:
I'd just reverse the order of the calls here and keep it that way.

I know it goes a bit against the rule of not putting code into a function that we only use once, but it's the same pattern we use for pool initialization, and here we're essentially doing the same thing. We're in the admin here so bytecode size is not an issue.